MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -28,35 +28,60 @@ public void Start(object hint = null)
 
         public virtual void ResetStart()
         {
+            Reset();
             Interlocked.Exchange(ref startHasCommenced, 0);
         }
+
+        protected virtual void Reset()
+        { }
     }
 
     public class MvxAppStart<TViewModel> : MvxAppStart
         where TViewModel : IMvxViewModel
     {
         protected readonly IMvxNavigationService NavigationService;
-        
-        public MvxAppStart(IMvxNavigationService navigationService)
+        protected readonly IMvxApplication Application;
+
+        public MvxAppStart(IMvxNavigationService navigationService, IMvxApplication application)
         {
             NavigationService = navigationService;
+            Application = application;
         }
 
         protected override void Startup(object hint = null)
+        {
+            ApplicationStartup();
+
+            NavigateToFirstViewModel(hint);
+        }
+
+        protected virtual void ApplicationStartup(object hint = null)
```

Comment:
Do we need this method? We could just call `Application.Startup(hint);` from `Startup `?

Reply:
I think it makes more sense now after this has been refactored